### PR TITLE
Add counters for an ElasticDL job

### PR DESCRIPTION
**Pull request by chunyang-wen**
Related to #1398 
This PR adds counters in `_TaskDispatcher`.

### CODE REVIEW

File: elasticdl/python/master/main.py [hunk 1]
```diff
@@ -6,8 +8,8 @@ def main():
     args = parse_master_args()
     master = Master(args)
     master.prepare()
-    master.run()
+    return master.run()
```
Comment (terrytangyuan) [inline elasticdl/python/master/main.py:11]: Why not exit directly from `master.run()`? Is this a better practice?

Comment (chunyang-wen) [reply elasticdl/python/master/main.py:11]: I think they are similar. But if we add `sys.exit(master.run())`, we have to make sure that this is the last function call. Using `sys.exit(main())` under the `__name__=="__main__"` works at the command line. Maybe it is more suitable.

Comment (brightcoder01) [reply elasticdl/python/master/main.py:11]: What's the returned value used for?

File: elasticdl/python/master/master.py [hunk 2]
```diff
@@ -146,6 +146,17 @@ def __init__(self, args):
         # Initialize instance manager
         self.instance_manager = self._create_instance_manager(args)
 
+        self._should_stop = False
+        self._exit_code = 0
+
+    def request_stop(self, err_msg=None):
```
Comment (brightcoder01) [inline elasticdl/python/master/master.py:152]: Which component will call this Api?

File: elasticdl/python/master/main.py (hunk 1 above)
Comment (chunyang-wen) [reply elasticdl/python/master/main.py:11]: To tell whether this command executes successfully or not.

File: elasticdl/python/master/master.py (hunk 2 above)
Comment (chunyang-wen) [reply elasticdl/python/master/master.py:152]: Any component. It provides a mechanism to stop master early. In another PR, I will use this API.

File: elasticdl/python/master/task_dispatcher.py [hunk 3]
```diff
@@ -30,6 +31,30 @@ def _info(self):
         )
 
 
+class JobCounters(object):
```
Comment (brightcoder01) [inline elasticdl/python/master/task_dispatcher.py]: JobCounter instead of JobCounters?

File: elasticdl/python/master/task_dispatcher.py [hunk 4]
```diff
@@ -30,6 +31,30 @@ def _info(self):
...
+    """Counters for job"""
+
+    def __init__(self, total_records=None, failed_records=None):
```
Comment (skydoorkai) [inline elasticdl/python/master/task_dispatcher.py]: Why not use `0` as argument default?

Comment (chunyang-wen) [reply elasticdl/python/master/task_dispatcher.py]: Good catch.

File: elasticdl/python/master/task_dispatcher.py (hunk 3 above)
Comment (chunyang-wen) [reply elasticdl/python/master/task_dispatcher.py]: Fixed

**Review by skydoorkai — APPROVED** [review]

**Pull request merged**